pkg/liqoctl/generate/handler.go

```diff
@@ -0,0 +1,125 @@
+	"github.com/liqotech/liqo/pkg/liqoctl/common"
+	"github.com/liqotech/liqo/pkg/liqonet/utils"
+)
+
+// HandleGenerateAddCommand outputs the liqoctl add command to use to add the target cluster.
+func HandleGenerateAddCommand(ctx context.Context, liqoNamespace, commandName string) {
+	// 7 attempts with 30 seconds sleep between one another
+	// for a total of 3 minutes
+	backoff := wait.Backoff{
+		Steps:    7,
+		Duration: 30 * time.Second,
+		Factor:   1.0,
+		Jitter:   0,
+	}
+	restConfig := common.GetLiqoctlRestConfOrDie()
+
+	cl, err := client.New(restConfig, client.Options{})
+	if err != nil {
+		klog.Fatalf(err.Error())
+	}
+	k8sClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		klog.Fatalf(err.Error())
+	}
+
+	localToken, err := auth.GetToken(ctx, cl, "liqo")
+	if err != nil {
+		klog.Fatalf(err.Error())
+	}
+
+	clusterID, err := utils.GetClusterID(k8sClient, consts.ClusterIDConfigMapName, liqoNamespace, backoff)
+	if err != nil {
+		klog.Fatalf("an error occurred while retrieving the clusterID: %s", err)
+	}
+
+	clusterConfig := &configv1alpha1.ClusterConfig{}
+	err = cl.Get(ctx, types.NamespacedName{
+		Name: consts.ClusterConfigResourceName,
+	}, clusterConfig)
+	if err != nil {
+		klog.Fatalf("an error occurred while retrieving the clusterConfig: %s", err)
+	}
+
+	authEP, err := getAuthSVCEndpoint(ctx, liqoNamespace, k8sClient)
+	if err != nil {
+		klog.Fatalf("an error occurred while retrieving the liqo-auth service: %s", err)
+	}
+
+	fmt.Printf("Use this command to peer with this cluster:\n\n")
+	fmt.Printf("%s\n", generateCommandString(commandName, authEP, clusterID, localToken, clusterConfig.Spec.DiscoveryConfig.ClusterName))
+}
+
+func generateCommandString(commandName, authEP, clusterID, localToken, clusterName string) string {
+	args := make(map[string]string)
+	args[add.AuthURLFlagName] = authEP
+	args[add.ClusterIDFlagName] = clusterID
+	args[add.ClusterTokenFlagName] = localToken
+	args[add.ClusterNameFlagName] = clusterName
+	command := []string{commandName, add.UseCommand, add.ClusterResourceName}
+	for flag, value := range args {
+		if value != "" {
+			command = append(
+				command,
+				"--"+flag,
+				value,
+			)
+		}
+	}
+	return strings.Join(command, " ")
+}
+
+func getAuthSVCEndpoint(ctx context.Context, liqoNamespace string, k8sClient kubernetes.Interface) (string, error) {
+	authSVC, err := k8sClient.CoreV1().Services(liqoNamespace).Get(ctx, consts.LiqoAuthenticationServiceName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	var authEP string
+	if authSVC.Spec.Type == corev1.ServiceTypeLoadBalancer && len(authSVC.Status.LoadBalancer.Ingress) > 0 {
+		authEP = getLBAuthService(authSVC)
+	} else if authSVC.Spec.Type == corev1.ServiceTypeNodePort {
+		nodes, err := k8sClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return "", err
+		}
+		authEP = getNodePortAuthService(authSVC, nodes)
+	}
+	return authEP, nil
+}
```

Comment:
It seems to me this logic is already present in the foreign-cluster-controller. Would generalize and reuse that, to avoid code duplication.

https://github.com/liqotech/liqo/blob/7616c1ccaa88ac28bef406d12daf9aab90910701/internal/discovery/foreign-cluster-operator/foreign-cluster-controller.go#L461-L498